### PR TITLE
Prometheus exporter for Layer 2 Lower MAC

### PR DESCRIPTION
**Pull request by marenz2569**
Things that this PR should implement:

- [x] counter for received bursts
- [x] counter for received bursts with/without lower mac decoding errors (crc16 ccitt)
- [x] counter for skipped bursts

### CODE REVIEW

**Comment by marenz2569** [comment]
- [ ] finish refactoring for the uplink w/o synchronization burst and given scrambling code

File: include/l2/access_assignment_channel.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include "l2/timebase_counter.hpp"
+#include <cstdint>
+#include <ostream>
+#include <vector>
+
+enum DownlinkUsage { CommonControl, Unallocated, AssignedControl, CommonAndAssignedControl, Traffic };
+
+class AccessAssignmentChannel {
+  public:
+    DownlinkUsage downlink_usage;
```
Comment (tanneberger) [inline include/l2/access_assignment_channel.hpp:21]: _

File: include/l2/broadcast_synchronization_channel.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#pragma once
+
+#include "burst_type.hpp"
+#include "l2/timebase_counter.hpp"
+#include <cstdint>
+#include <vector>
+
+class BroadcastSynchronizationChannel {
+  public:
+    uint8_t system_code{};
```
Comment (tanneberger) [inline include/l2/broadcast_synchronization_channel.hpp:18]: _

File: src/l2/lower_mac.cpp [hunk 3]
```diff
@@ -34,126 +48,83 @@ auto LowerMac::process(const std::vector<uint8_t>& frame, BurstType burst_type)
...
         uint8_t bb_des[30];
-        // XXX: upper_mac_->scrambling_code() may yield the wrong result!!! if the processing of the
-        // SynchronizationBurst is not done before this call, this may mean that the first packets are
-        // not decoded correctly.
-        descramble(bb.data(), bb_des, 30, upper_mac_->scrambling_code());
+        descramble(bb.data(), bb_des, 30, bsc.scrambling_code);
         std::vector<uint8_t> bb_rm(14);
         reed_muller_3014_decode(bb_des, bb_rm.data());
-        functions.push_back(std::bind(&UpperMac::process_AACH, upper_mac_, burst_type, bb_rm));
+        auto aach = AccessAssignmentChannel(burst_type, bsc.time, bb_rm);
```
Comment (tanneberger) [inline src/l2/lower_mac.cpp:87]: can we have a nicer name?

**Review by tanneberger — APPROVED** [review]
marenz a lot of variable names a just horrible

File: src/l2/lower_mac.cpp (hunk 3 above)
Comment (marenz2569) [reply src/l2/lower_mac.cpp:87]: well we can try, but sadly these are the ones which are used in the standard

Comment (tanneberger) [reply src/l2/lower_mac.cpp:87]: it just makes it incredibly hard to read that you always need to back to check what the different variables mean

**Pull request merged**